src/main/java/org/jenkinsci/plugins/workflow/steps/EnvStep.java

```diff
@@ -150,11 +153,37 @@
                     }
                 }
                 return b.toString();
+            } else if (overrides instanceof Map) {
+                return ((Map<?, ?>) overrides).keySet()
+                    .stream()
+                    .filter(e -> e instanceof String)
+                    .map(Object::toString)
+                    .collect(Collectors.joining(", "));
             } else {
                 return null;
             }
         }
 
+        @NonNull
+        @Override
+        public Map<String, Object> customInstantiate(@NonNull Map<String, Object> arguments) {
+            Map<String, Object> r = new HashMap<>(arguments);
+            final String key = "overrides";
+            Object overrides = r.get(key);
+            if (overrides instanceof Map) {
+                r.put(key, toKeyValueList((Map<?, ?>) overrides));
+            }
+            return r;
+        }
+
+        private static List<String> toKeyValueList(Map<?, ?> map) {
+            return map.entrySet().stream()
+                .filter(e -> e.getKey() instanceof String)
+                .collect(Collectors.toMap(e -> (String)e.getKey(), e -> e.getValue() == null ? "" : e.getValue().toString()))
+                .entrySet().stream()
+                .map(m -> m.getKey() + "=" + m.getValue())
```

Comment:
Why are you collecting a map and then restreaming it? Would seem to simpler to just

```java
.map(m -> (String) e.getKey() + "=" + (m.getValue() == null ? "" : m.getValue()))